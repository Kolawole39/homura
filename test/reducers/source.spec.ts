import { AnyAction } from 'redux';
import 'regenerator-runtime/runtime';
import {
  setCurrentSource,
  sourceReducer,
  State,
} from '../../app/features/source/sourceSlice';

describe('reducers', () => {
  describe('source', () => {
    it('should handle initial state', () => {
      expect(sourceReducer(undefined, {} as AnyAction)).toMatchSnapshot();
    });

    it('should handle setCurrentSource', () => {
      expect(
        sourceReducer({ activeId: null } as State, {
          type: setCurrentSource,
          payload: 1,
        })
      ).toMatchSnapshot();
    });

    it('should handle unknown action type', () => {
      expect(
        sourceReducer({ subscribeStep: null } as State, { type: 'unknown' })
      ).toMatchSnapshot();
    });
  });
});
