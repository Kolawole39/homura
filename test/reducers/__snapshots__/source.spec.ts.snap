// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers source should handle initial state 1`] = `
Object {
  "activeId": null,
  "fetchListStatus": 0,
  "list": Array [],
  "subscribeError": null,
  "subscribeLink": "",
  "subscribeName": "",
  "subscribeStatus": 0,
  "subscribeStep": null,
}
`;

exports[`reducers source should handle setCurrentSource 1`] = `
Object {
  "activeId": 1,
}
`;

exports[`reducers source should handle unknown action type 1`] = `
Object {
  "subscribeStep": null,
}
`;
